deepspeed/runtime/swap_tensor/async_swapper.py

```diff
@@ -37,7 +38,8 @@ def has_buffers(self):
 
     def add_buffers(self, buffer_list):
         assert len(self.all_buffers) == 0
-        assert all([buffer.is_pinned() for buffer in buffer_list])
+        assert all(
+            [get_accelerator().is_pinned(buffer) or get_accelerator().is_aligned(buffer) for buffer in buffer_list])
```

Comment:
Can you please explain why `is_aligned()` is an alternative for `is_pinned()`? We require `pinned()` buffers in the swapping logic because they are page-locked and so can be used for DMA. Are `aligned()` buffers guaranteed to work with DMA? If so, don't they have to be page-locked as well?